components/Common/CodeBox/index.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  DocumentDuplicateIcon,
+  CodeBracketIcon,
+} from '@heroicons/react/24/outline';
+import { useTranslations } from 'next-intl';
+import type { FC, PropsWithChildren, ReactNode } from 'react';
+import { Fragment, isValidElement, useMemo, useRef } from 'react';
+
+import Button from '@/components/Common/Button';
+import { useCopyToClipboard, useNotification } from '@/hooks';
+import { getCodeLanguageDisplayName } from '@/util/getCodeLanguageDisplayName';
+
+import styles from './index.module.css';
+
+type CodeBoxProps = {
+  language: string;
+};
+
+export const CodeBox: FC<PropsWithChildren<CodeBoxProps>> = ({
+  children,
+  language,
+}) => {
+  const ref = useRef<HTMLPreElement>(null);
+
+  const notify = useNotification();
+  const [, copyToClipboard] = useCopyToClipboard();
+  const t = useTranslations();
+
+  const onCopy = () => {
```

Comment:
I'm almost thinking that we should make the whole onCopy just a `prop` (callback) and let it be defined somewhere else. It is unpractical to make this a client-side component just because of the copy part of the codebox.

Then we can make MDXCodeBox the actual component that implements the `onCopy` (provides a valid callback) and the actual one to be "client-side". Since transformCode is done on the base CodeBox component, that would reduce the burden of the client-side of "parsing" and mapping the codebox for every CodeBox.